doc/src/explanation/solve_output.rst

```diff
@@ -0,0 +1,106 @@
+.. _solve_output:
+
+==========================
+Understanding Solve Output
+==========================
+
+The output of the `solve` function can seem very unwieldy since it may appear to
+arbitrarily return one of 6 different types of output (in addition to raising
+errors). The reasons for this are historical and are biased toward human
+interaction rather than programmatic use. The type of output will depend on the
+type of equation(s) (and how they are entered) and the number of symbols that
+are provided (and how they are provided).
+
+Since this output is controllable by setting `dict` or `set` to True
+(when not dealing with relational expressions) the following is an explanation for
+those that might prefer the *default* output but wish to understand better the
+conditions under which it is given.
```

Comment:
I think that it would be good to say explicitly here that for consistent output it is recommended to use `dict=True`.